### PR TITLE
[receiver/hostmetrics] Add documentation for mute_*_error configuration fields

### LINKED ISSUE #30779: Document the mute_process_*_error configuration options available in the process scraper
### Component(s)
receiver/hostmetrics

### Describe the issue you're reporting
At the time of writing, there are four `mute_process_*_error` configuration options, those are:

```
mute_process_name_error
mute_process_exe_error
mute_process_io_error
mute_process_user_error
```

However, there doesn't appear to be any documentation for these options. It's not clear what they are or why they exist - so they presently exist in the documentation unexplained.

I was looking to modify and test some existing process scraping configuration, and I wanted to understand if any of these options were applicable to the changes I was making, but I was unable to find any information that helped.

### PR DESCRIPTION
**Pull request by bsponge**
**Description:**
Added documentation for mute_*_error configuration fields and updated comments in the code.

**Link to tracking Issue:** 
Resolves #30779

**Testing:** not applicable

**Documentation:** Added missing documentation of mute_*_error configuration in hostmetricsreceiver

### CODE REVIEW

**Review by braydonk — CHANGES_REQUESTED** [review]
Hi @bsponge, thanks for opening a PR for this!

Unfortunately, the documentation can't be added where you have it here; note at the top of that file that it mentions it is autogenerated and thus shouldn't be edited.

What I'd suggest instead is adding it to the `README.md` in the root `hostmetricsreceiver` folder. In there, you'll see a section that shows the configuration options for every scraper. This can definitely be improved, but for now if you could add those descriptions directly below the `yaml` example that would be good for now. 

The rest of the PR looks good to me.

**Review by braydonk — APPROVED** [review]
Thanks, looks good to me! Will still need approval from dmitryax before merge.

**Review by crobert-1 — APPROVED** [review]
Thanks for your PR, always appreciate documentation enhancements!

**Comment by crobert-1** [comment]
_Hope you don't mind, I updated your PR description to [automatically close the linked issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#description-guidelines)_ 👍

File: .chloggen/add-mute-error-documentation.yaml
Comment (bsponge) [reply .chloggen/add-mute-error-documentation.yaml:1]: done

**Comment by braydonk** [comment]
unstale

**Comment by rogercoll** [comment]
Could we reopen this PR? This is still important documentation we should have in the `process` scrapper.

**Comment by crobert-1** [comment]
_Added `ready to merge` as it's been approved by an approver and code owner._

**Comment by braydonk** [comment]
@crobert-1 Maybe you'll know the answer to this. I'm not sure if as a codeowner in `.github/CODEOWNERS` my approval is supposed to count for codeowner approval according to GitHub (still has the grey checkmark here and in other PRs). This has been the case for a while but I wasn't sure if it was intentional or not, if I mixed something up with adding myself as a codeowner I can try to fix it.

**Comment by crobert-1** [comment]
> @crobert-1 Maybe you'll know the answer to this. I'm not sure if as a codeowner in `.github/CODEOWNERS` my approval is supposed to count for codeowner approval according to GitHub (still has the grey checkmark here and in other PRs). This has been the case for a while but I wasn't sure if it was intentional or not, if I mixed something up with adding myself as a codeowner I can try to fix it.

It's nothing wrong on your side, the green check mark is reserved for approvers and maintainers of the project, unrelated to code owners. Code owners are automatically added as reviewers on PRs, and we can also manually check the `.github/CODEOWNERS` file to see if someone is a code owner for a given component. (For reference, [here's a recent PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34063) where code owner approved and they also don't have a green check mark, just to confirm it's not you.)

I see you also requested a review from @dmitryax, but given this is only a documentation change, I assume we're good to merge. Is that accurate, or would you still prefer a review?

**Comment by braydonk** [comment]
Thanks for the clarification; I was used to other repos where approvals from CODEOWNERS were green check so I wasn't sure.

I thought merging would be blocked by Dmitrii's review; if it wouldn't be, then I imagine this is fine to merge.

**Comment by crobert-1** [comment]
> I thought merging would be blocked by Dmitrii's review; if it wouldn't be, then I imagine this is fine to merge.

One code owner approval is just fine by default, especially for something simple like this that's only documentation. That being said, if you're ever unsure about a change or want specific feedback from another code owner we can definitely block and ask for more input 👍

**Review by codeboten — APPROVED** [review]

**Pull request merged**
